src/dbus_api/pool.rs

```diff
@@ -85,16 +92,16 @@ fn create_filesystems(m: &MethodInfo<MTFn<TData>, TData>) -> MethodResult {
                         name,
                     )
                 })
-                .collect::<Vec<_>>();
-
-            return_message.append3(return_value, msg_code_ok(), msg_string_ok())
-        }
-        Err(err) => {
-            let (rc, rs) = engine_to_dbus_err_tuple(&err);
-            return_message.append3(default_return, rc, rs)
+                .collect::<Vec<_>>()
         }
+        None => Vec::new(),
     };
-    Ok(vec![msg])
+
+    Ok(vec![return_message.append3(
```

Comment:
Tip: When you change the values returned on the D-Bus API you have to change the spec, too. See around line 395 in this file for the code that declares this for this method. This uses the D-Bus type signature, at this time is is "a(os)" meaning an array of tuple of dbus object path * string. You'll want in this case, to make it "(ba(os))", I think. Whether the value returned on the D-Bus conforms to the declared D-Bus return type is not statically checked, so it's necessary to be very careful about  consistency of return values.

This is where you'll also update the test infrastructure: tests/client-dbus/src/stratisd_client_dbus/_data.py. This contains the introspection xml. When you change the spec in the stratisd D-Bus layer, you must change this correspondingly.

Reply:
This all should be resolved in my revision commit.